find.go

```diff
@@ -175,12 +193,24 @@ func (s *server) find(w http.ResponseWriter, r *http.Request) {
 	}
 	ctx := r.Context()
 
+	if mh == nil {
+		// mh should only be nil for batch find POST request.
+		// Sanity check it and return 500 if it is.
+		log.Error("multihash must not be nil for single find requests; mh should only be nil for batch find POST request.")
+		http.Error(w, "", http.StatusInternalServerError)
+		return
+	}
+
 	// Use NDJSON response only when the request explicitly accepts it. Otherwise, fallback on
 	// JSON unless only unsupported media types are specified.
 	switch {
 	case acc.ndjson:
-		s.doFindNDJson(ctx, w, r.Method, findMethodOrig, r.URL)
+		s.doFindNDJson(ctx, w, findMethodOrig, r.URL, false, mh)
 	case acc.json || acc.any || !acc.acceptHeaderFound:
+		if s.translateNonStreaming {
+			s.doFindNDJson(ctx, w, findMethodOrig, r.URL, true, mh)
+			return
+		}
 		// In a case where the request has no `Accept` header at all, be forgiving and respond with
 		// JSON.
 		rcode, resp := s.doFind(ctx, r.Method, findMethodOrig, r.URL, rb)
```

Comment:
do we want to maintain both `doFind` and `doFindNDJson`? with the `translateNonStreaming` bool in `doFindNDJson` could we use that same code path for all backend lookups?

Reply:
I agree, now that all the deployed backends support streaming we should default to it in indexstar. 
There is a lot that I'd like to refactor here, which hope to approach in separate PRs.

Reply:
sounds good :+1: